frontend/docs/patterns.md

```diff
@@ -7,61 +7,108 @@ follow these patterns unless a specific use case calls for something else. These
 should be discussed within the team and documented before merged.
 
 ## Table of contents
-  - [Typing](#typing)
-  - [Utilities](#utilities)
-  - [Components](#components)
-  - [React Hooks](#react-hooks)
-  - [React Context](#react-context)
-  - [Fleet API Calls](#fleet-api-calls)
-  - [Page Routing](#page-routing)
-  - [Styles](#styles)
-  - [Icons and Images](#icons)
-  - [Other](#other)
+
+- [Typing](#typing)
+- [Utilities](#utilities)
+- [Components](#components)
+- [React Hooks](#react-hooks)
+- [React Context](#react-context)
+- [Fleet API Calls](#fleet-api-calls)
+- [Page Routing](#page-routing)
+- [Styles](#styles)
+- [Icons and Images](#icons-and-images)
+- [Other](#other)
 
 ## Typing
+
 All Javascript and React files use Typescript, meaning the extensions are `.ts` and `.tsx`. Here are the guidelines on how we type at Fleet:
 
 - Use *[global entity interfaces](../README.md#interfaces)* when interfaces are used multiple times across the app
 - Use *local interfaces* when typing entities limited to the specific page or component
-- Local interfaces for page and component props
-
-  ```typescript
-  // page
-  interface IPageProps {
-    prop1: string;
-    prop2: number;
-    ...
-  }
 
-  // Note: Destructure props in page/component signature
-  const PageOrComponent = ({
-    prop1,
-    prop2,
-  }: IPageProps) => {
+### Local interfaces for page, widget, or component props
 
-    return (
-      // ...
-    );
-  };
-  ```
+```typescript
+// page
+interface IPageProps {
+  prop1: string;
+  prop2: number;
+  ...
+}
+
+// Note: Destructure props in page/component signature
+const PageOrComponent = ({ prop1, prop2 }: IPageProps) => {
+  // ...
+};
+```
+
+### Local states with types
 
-- Local states
 ```typescript
+// Use type inference when possible.
 const [item, setItem] = useState("");
+
+// Define the type in the useState generic when needed.
+const [user, setUser] = useState<IUser>()
 ```
 
-- Fetch function signatures (i.e. `react-query`)
+### Fetch function signatures (i.e. `react-query`)
+
 ```typescript
-useQuery<IHostResponse, Error, IHost>(params)
+// include the types for the response, error.
+const { data } = useQuery<IHostResponse, Error>(
+  'host',
+  () => hostAPI.getHost()
+)
+
+
+// include the third host data generic argument if the response data and exposed data are different.
+// This is usually the case when we use the `select` option in useQuery.
+
+// `data` here will be type IHostProfiles
+const { data  } = useQuery<IHostResponse, Error, IHostProfiles>(
+  'host',
```

Comment:
Did we want to define any query key patterns here?

Reply:
this section is specifically for typings so I think we should add that to a different section.